laa_court_data_api_app/routers/hearing_summaries.py

```diff
@@ -71,7 +71,7 @@ def map_hearing_summaries(prosecution_case_results: list[ProsecutionCases]):
 def map_defendants_from_guids(defendant_ids: list[str], defendant_summaries: list[DefendantSummary]):
     defendant_list = []
     for defendant_id in defendant_ids:
-        filtered_defendant_summaries = filter(lambda defendants: str(defendants.id) == defendant_id,
+        filtered_defendant_summaries = filter(lambda defendants, d_id=defendant_id: str(defendants.id) == d_id,
```

Comment:
which code smell does this fix and what's the benefit of it?

Reply:
Code smells can be found here - https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&id=ministryofjustice_laa-court-data-api